common/txmgr/types/storage_service.go

```diff
@@ -0,0 +1,95 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// NEWTX, TX, TXATTEMPT will be converted from generic types to structs at a future date to enforce design and type checks
+type TxStorageService[ADDR any, CHAINID any, HASH any, NEWTX any, R any, TX any, TXATTEMPT any, TXID any, TXMETA any] interface {
+	CheckEthTxQueueCapacity(fromAddress ADDR, maxQueuedTransactions uint64, chainID CHAINID, opts ...any) (err error)
+	CountUnconfirmedTransactions(fromAddress ADDR, chainID CHAINID, opts ...any) (count uint32, err error)
+	CountUnstartedTransactions(fromAddress ADDR, chainID CHAINID, opts ...any) (count uint32, err error)
+	CreateEthTransaction(newTx NEWTX, chainID CHAINID, opts ...any) (tx TX, err error)
+	DeleteInProgressAttempt(attempt TXATTEMPT, opts ...any) error
+	EthTransactions(offset, limit int, opts ...any) ([]TX, int, error)
+	EthTransactionsWithAttempts(offset, limit int, opts ...any) ([]TX, int, error)
+	// EthTxAttempts(offset, limit int, opts ...any) ([]TXATTEMPT, int, error)
+	FindEthReceiptsPendingConfirmation(ctx context.Context, blockNum int64, chainID CHAINID, opts ...any) (receipts []ReceiptPlus[R], err error)
```

Comment:
I don't fully understand what is going on with these `any` opts. What does this achieve? Why would we give up compile time type safety?

Reply:
Curious as to @samsondav 's opinion since we agreed that qopts was a good direction for the eth orm's extensibility and standardization. How would you suggest generalizing that? Off the top of my head if we want compile-time type safety, then we can introduce a generic parameter for database/queryer options. Each ORM is specific to a database, so it figures it should have database/queryer-specific options?

Reply:
Why can't it just take `qopts ...pg.QOpt` like before?